File: LeetcodeQuestions/count_primes-204.rb
# @param {Integer} n
# @return {Integer}
def count_primes(n)
    primes = Array.new(n).fill(false)
    count = 0
    
    (2...primes.length).each do |i|
        if primes[i] == false
            j = i
            while j * i < primes.length
                primes[i * j] = true
                j += 1
            end
        end
    end
    
    (2...primes.length).each do |i|
        if primes[i] == false
            count += 1
        end
    end
    count
end

File: LeetcodeQuestions/countPrimes-204.js
function countPrimes(n) {
    let count = 0;
    let primes = new Array(n).fill(false);
    for (let i = 2; i * i < primes.length; i++) {
        if (!primes[i]) {
            for (let j = i; j * i < primes.length; j++) {
                primes[i * j] = true;
            }
        }
    }

    for (let i = 2; i < primes.length; i++) {
        if (!primes[i]) count++;
    }
    return count;
}

console.log(countPrimes(10))
console.log(countPrimes(100))